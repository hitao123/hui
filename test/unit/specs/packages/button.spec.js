import { shallowMount } from '@vue/test-utils';
import Button from '../../../../packages/button';

describe('Loading component', () => {
  it('should Loading as Circular', () => {
    const warpper = shallowMount(Button);
    warpper.find('svg');
    expect(warpper.find('svg')).toBeDefined();
  });

  it('should Loading as Spinner', () => {
    const warpper = shallowMount(Button);
    expect(warpper.find('svg')).toBeDefined();
  });
});
