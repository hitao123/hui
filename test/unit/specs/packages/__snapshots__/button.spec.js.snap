// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should show Button 1`] = `
<button class="h-button h-button--default h-button--normal h-button--plain">
  <!----> <span class="h-button__text">Button</span></button>
`;

exports[`Button component should show Button Demo 1`] = `
<div class="button-demo">
  <div>
    <h3 class="h3">type</h3> <button class="h-button h-button--default h-button--normal">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--primary h-button--normal">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--warning h-button--normal">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--danger h-button--normal">
      <!----> <span class="h-button__text">Button</span></button>
  </div>
  <div>
    <h3 class="h3">plain</h3> <button class="h-button h-button--default h-button--normal h-button--plain">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--primary h-button--normal h-button--plain">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--warning h-button--normal h-button--plain">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--danger h-button--normal h-button--plain">
      <!----> <span class="h-button__text">Button</span></button>
  </div>
  <div>
    <h3 class="h3">size</h3> <button class="h-button h-button--default h-button--large">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--primary h-button--normal h-button--plain">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--warning h-button--small">
      <!----> <span class="h-button__text">Button</span></button> <button class="h-button h-button--danger h-button--mini">
      <!----> <span class="h-button__text">Button</span></button>
  </div>
  <div>
    <h3 class="h3">disabled</h3> <button class="h-button h-button--default h-button--large h-button--disabled">
      <!----> <span class="h-button__text">Button</span></button>
  </div>
  <div>
    <h3 class="h3">loading</h3> <button class="h-button h-button--default h-button--normal h-button--loading">
      <div class="h-loading h-loading--circular h-loading--black" style="width:20px;height:20px;"><span class="h-loading__spinner h-loading__spinner--circular"> <svg viewBox="25 25 50 50" class="h-loading__circular"><circle cx="50" cy="50" r="20" fill="none"></circle></svg></span></div>
      <span class="h-button__text"></span>
    </button>
  </div>
</div>
`;
